File: Diagrammatize.Design.Pattern/code/cxx/State/A4/UrgentState.cpp
#include "UrgentState.hpp"
#include "Context.hpp"

UrgentState::UrgentState() {
}

State* UrgentState::getInstance() {
    static UrgentState singleton;
    return &singleton;
}

void UrgentState::doClock(Context* context, int hour) {
}

void UrgentState::doUse(Context* context) {
    context->callSecurityCenter("紧急:紧急时使用金库！");
}

void UrgentState::doAlarm(Context* context) {
    context->callSecurityCenter("按下警铃(紧急时)");
}

void UrgentState::doPhone(Context* context) {
    context->callSecurityCenter("正常通话(紧急时)");
}

std::string UrgentState::toString() const {
    return "[紧急时]";
}

File: Diagrammatize.Design.Pattern/code/python/State/A4/UrgentState.py
from State import State

class UrgentState(State):
    _singleton = None

    def __init__(self):
        print("create UrgentState")
        pass

    @classmethod
    def getInstance(cls):
        if not cls._singleton:
            cls._singleton = cls()
        return cls._singleton

    def doClock(self, context, hour):
        pass

    def doUse(self, context):
        context.callSecurityCenter("紧急:紧急时使用金库！")

    def doAlarm(self, context):
        context.callSecurityCenter("按下警铃(紧急时)")

    def doPhone(self, context):
        context.callSecurityCenter("正常通话(紧急时)")

    def __str__(self):
        return "[紧急时]"


